### ISSUE TITLE
Package request: moondeck-buddy

### ISSUE THREAD

**Issue by keenanweaver** [NixOS/nixpkgs#354802]
**Project description**

A server-side part of the [MoonDeck](https://github.com/FrogTheFrog/moondeck) plugin for the SteamDeck.

**Metadata**

* homepage URL: n/a
* source URL: https://github.com/FrogTheFrog/moondeck-buddy
* license: GPLv3
* platforms: linux

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by ccalhoun1999** [comment]
I tried getting this working a bit ago in a flake but gave up. There seem to be some dependencies that I'm either too incompetent to find, or simply don't exist yet in nixpkgs. Would love if we could get this though.

https://discourse.nixos.org/t/problem-building-a-package-using-cmake-depending-on-proc-procps-in-flake/57201/5

**Comment by keenanweaver** [comment]
Yeah, the appimage has never worked for me on NixOS. I've tried to package it a few times before as well and didn't have the patience to figure it out.

**Comment by nuko2MB** [comment]
I packaged this in my flake a while ago. I don't use it anymore, but it worked when I tried it. As long as you open the required port (default: 5999). The latest release (1.6.2) only builds on 24.05, upgraded dependencies broke it. This was recently fixed upstream, but there is no stable release yet.

Needs to be built with `qt6Packages.callPackage`.


```nix
{
  stdenv,
  lib,
  fetchFromGitHub,
  qt6,
  ninja,
  cmake,
  wrapQtAppsHook,
  procps,
  libXrandr,
  fuse,
  xcb-util-cursor,
  steam,
}:
let
  pname = "moondeck-buddy";
  #version = "1.6.2";
  version = "ffb52260e08f6aa354c147ffe957f323f386d737";
  src = fetchFromGitHub {
    owner = "FrogTheFrog";
    repo = "moondeck-buddy";
    #rev = "v${version}";
    rev = version;
    sha256 = "sha256-wPkMryosIltohWssVHbpBe+8WlHWJA4Tcb5AXBZGIHo=";
    fetchSubmodules = true;
  };
in
stdenv.mkDerivation {
  inherit pname version src;
  nativeBuildInputs = [
    ninja
    cmake
    wrapQtAppsHook
  ];
  buildInputs = [
    qt6.qtbase
    qt6.qthttpserver
    libXrandr
    procps
    steam
    qt6.qtwebsockets
    fuse
    xcb-util-cursor
  ];

  postPatch = ''
    substituteInPlace src/lib/os/linux/steamregistryobserver.cpp \
        --replace-fail /usr/bin/steam ${lib.getExe steam};
  '';
}
```

**Comment by keenanweaver** [comment]
That's awesome! Thanks for sharing. I'll probably wait for a new stable release before futzing with it again.

**Comment by redxtech** [comment]
I opened a PR to add this (#375287) a couple days ago, hopefully it doesn't take too long to get merged!

@nuko2MB thanks for the snippet! I was handling the steam binary issue by setting the path in the settings file, so patching the file is a much better way to go about this, I've just updated my PR to include it!

**Comment by nuko2MB** [comment]
@redxtech 


Using `/usr/bin/env steam` (or just `steam`) is possible but I don't think it's preferred. This also requires patching out a call to `QFile::exists`. This works but will break error messages when setting the path via config. And, while unlikely, it could cause issues in future versions as the code is expecting an absolute file path.

Not patching it is always an option, but since the code expects an absolute path it would require something like a home-manager module to properly set the config value to the store path.

\
I believe the best option is to patch using an absolute nix store path. The only downside is that this pulls in steam as a dependency when it may not be needed. Such as using flatpak steam or using a non nixos system. 

This seems somewhat niche to me and if desired could be mitigated with `useNixSteam ? true` and `lib.optional`, but I would ask a maintainer what is preferred.

---

While I'm here, in your derivation `cmake` and `ninja`  should be moved to `nativeBuildInputs`. Then remove `cmakeFlags`. I also don't think `pkg-config` is required, but may be wrong.

**Comment by redxtech** [comment]
> Using `/usr/bin/env steam` (or just `steam`) is possible but I don't think it's preferred. This also requires patching out a call to `QFile::exists`. This works but will break error messages when setting the path via config. And, while unlikely, it could cause issues in future versions as the code is expecting an absolute file path.

This makes sense. I'll leave it to `getExe steam`.

> Not patching it is always an option, but since the code expects an absolute path it would require something like a home-manager module to properly set the config value to the store path.

Yes definitely! Once this is merged, I can PR home-manager to add a module for this.

> I believe the best option is to patch using an absolute nix store path. The only downside is that this pulls in steam as a dependency when it may not be needed. Such as using flatpak steam or using a non nixos system.
> 
> This seems somewhat niche to me and if desired could be mitigated with `useNixSteam ? true` and `lib.optional`, but I would ask a maintainer what is preferred.

This makes sense. I've updated my PR to add the `useNixSteam` arg, and `lib.optionalString` to only patch the bin path when it is true.
 
> While I'm here, in your derivation `cmake` and `ninja` should be moved to `nativeBuildInputs`. Then remove `cmakeFlags`. I also don't think `pkg-config` is required, but may be wrong.

Moved those deps, and yeah it looks like `pkg-config` isn't required.

Thanks for the help! If you're at all interested, I can add you to the maintainers list on the package, so you can help approve and review updates for the package.
